pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [column is referenced by the primary key]

pkg/ccl/testccl/workload/schemachange/schemachange_test.TestWorkload [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18047462?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18047462?buildTab=artifacts#/) on master @ [f9a60821a1d13b7dc6f5f7fe249eafaba0233883](https://github.com/cockroachdb/cockroach/commits/f9a60821a1d13b7dc6f5f7fe249eafaba0233883):


```
    "query": "WITH tab_json AS (SELECT crdb_internal.pb_to_json('desc', descriptor)-\u003e'table' AS t FROM system.descriptor WHERE id = $1::REGCLASS), columns_json AS (SELECT json_array_elements(t-\u003e'columns') AS c FROM tab_json), columns AS (SELECT c-\u003e\u003e'computeExpr' AS generation_expression, c-\u003e\u003e'name' AS column_name, c-\u003e\u003e'id' AS ordinal FROM columns_json) SELECT generation_expression FROM columns WHERE generation_expression IS NOT NULL",
    "queryArgs": [
     "public.table_w0_6"
    ],
    "result": []
   }
  ],
  "previousStatements": [
   "ALTER TABLE public.table_w0_6 DROP COLUMN col6_w0_7"
  ],
  "usesDeclarativeSchemaChanger": true
 }
}
    schema_change_external_test.go:131: 
        	Error Trace:	pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:131
        	Error:      	Received unexpected error:
        	            	***UNEXPECTED ERROR; Received an unexpected execution error.: ERROR: column "col6_w0_7" is referenced by the primary key (SQLSTATE 42P10)
        	            	(1) forced error mark
        	            	  | "fatal error when running txn"
        	            	  | github.com/cockroachdb/errors/withstack/*withstack.withStack::
        	            	Wraps: (2)
        	            	Wraps: (3) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*opStmt).executeStmt
        	            	  | 	pkg/workload/schemachange/operation_generator.go:3234
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	pkg/workload/schemachange/schemachange.go:503
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).run
        	            	  | 	pkg/workload/schemachange/schemachange.go:600
        	            	  | pkg/ccl/testccl/workload/schemachange/schemachange_test_test.TestWorkload.TestWorkload.func4.func6
        	            	  | 	pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:120
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	external/org_golang_x_sync/errgroup/errgroup.go:78
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_arm64.s:1222
        	            	Wraps: (4) ***UNEXPECTED ERROR; Received an unexpected execution error.
        	            	Wraps: (5) ERROR: column "col6_w0_7" is referenced by the primary key (SQLSTATE 42P10)
        	            	Error types: (1) *markers.withMark (2) *schemachange.ErrorState (3) *withstack.withStack (4) *errutil.withPrefix (5) *pgconn.PgError
        	Test:       	TestWorkload
Schema Workload Stats
Total Schema Statements Executed = 58
Total Schema Statements Succeeded = 58
Total Schema Statement Expected Failures = 0
Total Transactions Committed = 36
Total Transactions Rolled Back = 27
Total Transactions Executed = 63
    schema_change_external_test.go:96: backup, tracing data, and system table dumps in /artifacts/tmp/_tmp/67537770b84c3a4ea177539c1e46bd43/logTestWorkload2251999444
    panic.go:626: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/67537770b84c3a4ea177539c1e46bd43/logTestWorkload2251999444
--- FAIL: TestWorkload (20.29s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #136308 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot P-3 T-sql-foundations branch-release-24.3.0-rc]
- #136305 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot P-3 T-sql-foundations branch-release-24.3.1-rc]
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWorkload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-45264

--- a/pkg/workload/schemachange/error_screening.go
+++ b/pkg/workload/schemachange/error_screening.go
@@ -1445,14 +1445,14 @@ WITH descriptors AS (
                           FROM descriptors
                        )
                  WHERE (mut->'primaryKeySwap') IS NOT NULL
-            )
 		 -- Check for declarative primary key swaps, which will appear as
 		 -- as new primary indexes with different key columns
-     UNION SELECT count(*) > 0
+     UNION SELECT p
              FROM primaryindex AS pk, mutations AS mut
             WHERE m->'index'->'encodingType' = '1'::JSONB
                   AND m->'index'->'keyColumnIds'
-                    != p->'keyColumnIds';
+                    != p->'keyColumnIds'
+			);
 		`,
 		tableName.String(),
 	)

